Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -281,6 +281,23 @@ def check_for_beta_xquartz
           We recommend only installing stable releases of XQuartz.
         EOS
       end
+
+      def check_for_sip_deleted_header_file_installed_by_xcode
+        # Starting in El Capitan the upgrade of macOS lead to the deletion of the
+        # /usr/include directory due to SIP.
+        return if MacOS::CLT.installed? && File.file?("/usr/include/zlib.h")
```

Comment:
`xcode-select --install` just installs the CLT so if that's the fix then they aren't currently installed correctly. I mean that check is currently bad because it's not checking for a working CLT.